src/core/coap/coap.cpp

```diff
@@ -398,7 +385,7 @@ Message *Coap::CopyAndEnqueueMessage(const Message &aMessage, uint16_t aCopyLeng
     uint32_t alarmFireTime;
 
     // Create a message copy of requested size.
-    VerifyOrExit((messageCopy = aMessage.Clone(aCopyLength)) != NULL, error = OT_ERROR_NO_BUFS);
+    VerifyOrExit((messageCopy = static_cast<Message *>(aMessage.Clone(aCopyLength))) != NULL, error = OT_ERROR_NO_BUFS);
```

Comment:
Is this `static_cast` necessary with the introduction of `ot::Coap::Message::Clone()` methods?